### PR TITLE
Split benchmarking into public API and internal

### PR DESCRIPTION
**Pull request by witchel**
Put file and transaction benchmarking, which rely on "public" APIs in tsuba_bench.cpp.  Leave S3 internal benchmarking in s3_bench.cpp.  This mostly moves code around, but I made some effort to increase the C++-i-ness of the code.  The hope is that tsuba_bench can be useful for things like assessing the performance of the azure back end.

### CODE REVIEW

File: libtsuba/tests/bench_utils.h [hunk 1]
```diff
@@ -26,19 +27,60 @@ inline struct timespec timespec_sub(struct timespec time,
 }
 
 inline int64_t timespec_to_us(struct timespec ts) {
-  return ts.tv_sec * 1'000'000 + ts.tv_nsec / 1'000;
+  return ts.tv_sec * 1'000'000 + ts.tv_nsec / 1'000; // '
```
[the remark this answers is not in the document]
Comment (witchel) [reply libtsuba/tests/bench_utils.h:30]: I really like the ability to add ' into numeric constants.  But my version of cc-mode in emacs likes to count single quote characters.  If there are an odd number, it worries that I haven't closed a character constant.  That comment prevents my emacs from coloring the rest of the buffer as a string.

**Comment by witchel** [comment]
> Seems fine. Other parts of the code base are using [google benchmark](https://github.com/google/benchmark) I don't think its worth porting there atm, just putting it on the proverbial radar.

Nice.  I looked at it a little when I started this, but it seemed like more than I wanted to take on at the time.  Maybe we can start filtering it in.  Splitting off tsuba_bench makes it a softer target.

I know that real programmers use `llvm::cl::ParseCommandLineOptions`, but I just haven't taken the time yet.

File: libtsuba/tests/bench_utils.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (tylershunt) [reply libtsuba/tests/bench_utils.h:30]: word. makes sense.

**Comment by tylershunt** [comment]
> real programmers use `llvm::cl::ParseCommandLineOptions`

* real programmers that want to keep llvm as a dependency use `llvm::cl::ParseCommandLineOptions`

**Pull request merged**